src/gfw/common/beam/transforms/read_matching_avro_files.py

```
@@ -133,12 +134,12 @@ def _decode_records(self, record: dict) -> dict:
 
     def is_path_in_range(self, path: str) -> bool:
         """Checks if a path containing a datetime is within the provided datetime range."""
-        dt = get_datetime_from_string(path)
-        if dt is None:
-            logger.error(
-                f"Couldn't extract datetime from path: {path} using regex: {self.DATETIME_REGEX}"
+        try:
+            dt = datetime_from_string(
+                path, date_format=self._date_format, time_format=self._time_format
             )
-
+        except ValueError as e:
+            logger.error(f"Couldn't extract datetime from path: {e}")
```

Should we raise an exception here instead of just logging and return False? 🤔